### ISSUE TITLE
Metro minify changes code behavior

### ISSUE THREAD

**Issue by yulanggong** [facebook/metro#562]
**Do you want to request a *feature* or report a *bug*?**

report a *bug*

**What is the current behavior?**

`entry.js`
```js
function bnpInvDigit(b, DV) {
    if ((b & 1) == 0) {
        return 0;
    }
    var a = b & 3,
        a = (a * (2 - (b & 15) * a)) & 15,
        a = (a * (2 - (b & 255) * a)) & 255,
        a = (a * (2 - (((b & 65535) * a) & 65535))) & 65535,
        a = (a * (2 - ((b * a) % DV))) % DV;
    return a > 0 ? DV - a : -a;
}

console.log(bnpInvDigit(3, 3)); // 2
console.log(bnpInvDigit(3, 6)); // 5
```

```sh
metro build entry.js -O out.js
```
The codes passed to minifier in `node_modules/metro/src/JSTransformer/worker.js`: 
```js
__d(function (g, r, i, a, m, e, d) {
  "use strict";

  function bnpInvDigit(b, DV) {
    if ((b & 1) == 0) {
      return 0;
    }

    var _a3 = b & 3,
        // from here caused the bug
        a = _a3 * (2 - (b & 15) * _a3) & 15,
        a = _a3 * (2 - (b & 255) * _a3) & 255,
        a = _a3 * (2 - ((b & 65535) * _a3 & 65535)) & 65535,
        // end of the bug
        _a3 = _a3 * (2 - b * _a3 % DV) % DV;

    return _a3 > 0 ? DV - _a3 : -_a3;
  }

  console.log(bnpInvDigit(3, 3));
  console.log(bnpInvDigit(3, 6));
});
```

The primary part of the result:

```js
function n(n, o) {
    if (0 == (1 & n)) return 0;
    var t;
    t = 3 & n;
    return (t = (t * (2 - ((n * t) % o))) % o) > 0 ? o - t : -t;
}
console.log(n(3, 3)), // -0
console.log(n(3, 6)); // 3
```

**If the current behavior is a bug, please provide the steps to reproduce and a minimal repository on GitHub that we can `yarn install` and `yarn test`.**

as above

**What is the expected behavior?**

do it right

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**

* Metro 0.59.0

**Comment by yulanggong** [comment]
This maybe a bug of `uglify-es`, but the `uglify-es` project is no longer maintained.

**Comment by yulanggong** [comment]
https://github.com/facebook/metro/blob/ccba71d12663b53190005feb7c0af953d24e8f7e/packages/metro-transform-plugins/src/normalizePseudoGlobals.js#L84

This line caused the bug, maybe a `babel` bug.

**Comment by yulanggong** [comment]
bug fixed in `babel v7.10.0`
https://github.com/babel/babel/pull/11595
